frontend/src/components/profile/profile.tsx

```diff
@@ -1,14 +1,29 @@
-import { useEffect, useAppDispatch, useAppSelector } from 'hooks/hooks';
+import {
+  useEffect,
+  useAppDispatch,
+  useAppSelector,
+  useNavigate,
+} from 'hooks/hooks';
 import { profileActions } from 'store/actions';
 
 import Tabs from './tabs/tabs';
 import Header from './header/header';
 import ProfileMain from './profile-main/profile-main';
 import './styles.scss';
-
 const ProfileInfo: React.FC = () => {
   const dispatch = useAppDispatch();
   const { user, isLoading } = useAppSelector((state) => state.profile);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (user) {
```

Comment:
try to simplify this logic. Very hard to understand what is going on here. You use `if` then `ternary operator` then `ternary operator`